WebAssembly/Runtime/UnmanagedMemory.cs

```diff
@@ -96,7 +170,16 @@ public uint Grow(uint delta)
 
                 var newCurrent = oldCurrent + delta;
                 var newSize = newCurrent * Memory.PageSize;
-                this.Start = Marshal.ReAllocHGlobal(this.Start, new IntPtr(newSize));
+                if (this.Start == default)
```

Comment:
The only condition where start could be `default` is if the object was disposed.  So, the correct thing to do here is throw an `ObjectDisposedException`.

Reply:
I added the throw, but you should note that you have the entire thing wrapped in a `try { ... } catch {}` swallowing any exceptions. This also causes failures from memory_grow lines 19 to 60, so it's safe to say that those are due to prematurely disposed memory or uses of uninitialized memory.

Reply:
_Interesting._  That suggests there's a problem with the initialization code inside/generated by the compiler.  If that's the only thing preventing all those spec tests from passing, I'll definitely want to look into that soon for an easy compliance improvement.